src/content/reference/cluster-upgrades/index.md

```diff
@@ -0,0 +1,190 @@
+To get information on all available releases, use the [`gsctl list releases`](/reference/gsctl/list-releases/) command.
+
+### Release lifecycle
+
+**We hold two different major releases** available at any time to chose from when creating a new cluster, which means that you have two different Kubernetes minor versions to chose from.
+
+Whenever a new Kubernetes minor version is released by the Kubernetes project, we aim to make that version available in a new major release of our stack within 30 days from the Kubernetes release.
+
+Once we publish a new major release, we deprecate the oldest major release.
+This means that no new clusters can be created using that old release version.
+Existing clusters, however, are not affected.
+
+**Both patch and minor upgrades** can be rolled out at any time by Giant Swarm without your interaction. Currently, this happens in coordination with your administrators and with a notice to your developers.
+
+When a **new major release** becomes available, we inform you, but leave scheduling of the upgrade to you. This gives you the control to decide if and when it is time for you to upgrade, potentially updating workloads first. These upgrades are also acommpanied or even triggered by Giant Swarm staff, to ensure we have a close eye on the upgrade process and the uptime of your workloads.
+
+Every new minor Kubernetes release, which comes with a major Giant Swarm release, is tested for conformance using the CNCF [conformance test suite](https://github.com/cncf/k8s-conformance).
+In addition, every release, from patch to major, undergoes automated integration testing.
+
+## How upgrades work
+
+All levels of upgrades, patch, minor, or major, are happening at runtime.
+Your Kubernetes workloads should continue to work as expected (given they meet a few requirements, as explained further below).
+
+In particular this means:
+
+- Nodes will be drained, then stopped, then recreated.
+- As a consequence of drainging, Pods running on a node will be rescheduled to other nodes. 
+- Once the master node will is taken down and recreated, the Kubernetes API will be unavailable for a short time.
+
+**Note**: Currently tenant clusters have one master node each. We have plans on our roadmap to allow for multiple master nodes, keeping the Kubernetes API accessible during an upgrade and increasing the resiliance in case of a machine failure.
+
+### Provider-specific details for AWS
+
+AWS resources are managed by the `aws-operator` component through a nested CloudFormation stack. For a tenant cluster upgrade, the CloudFormation stacks is updated, in several steps.
+
+The master node re-creation is started first. Meanhile, the recreation of worker nodes starts, where the all worker nodes are recreated in batches. During the upgrade, **up to 33 percent of the worker nodes can be unavailable**.
```

Comment:
The ratio of one third is actually a problem for bigger clusters. Lets note that we are aware of it and have ideas on the roadmap to tackle this eventually.